### PR TITLE
Quarterly report addition

### PR DESCRIPTION
**Pull request by HelenCampbell**
The addition of a script to discover quarterly metrics of total PRs created, merged and closed. As it'd only be run once every quarter a couple of things are hardcoded - puppet member list and start and end boundaries of the quarter.

### CODE REVIEW

File: quarterly_report.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  #Retrieves all PRs for the repo
+  pr_cache = (util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'open'}, filter=[]))
+  pr_cache.concat(util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'closed'}, filter=[]))
+
+  pr_cache.each do |pr|
+    all_pulls.push (pr[:pull])
+  end
+end
+
+puppet_members = ["bmjen", "HelenCampbell", "hunner", "DavidS", "tphoney", "jonnytpuppet", "garethr"]
```
Comment (DavidS) [inline quarterly_report.rb]: +gregohardy, +mentat ?

File: quarterly_report.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  opts.on('--puppetlabs', 'Select Puppet Labs\' modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = '^puppetlabs-'
+  }
+  opts.on('--puppetlabs-supported', 'Select only Puppet Labs\' supported modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = OctokitUtils::SUPPORTED_MODULES_REGEX
+  }
+  opts.on('--community', 'Select community modules') {
+    options[:namespace] = 'puppet-community'
```
Comment (DavidS) [inline quarterly_report.rb]: voxpupuli, not puppet-community; should proabably fixed in all scripts - or, as discussed, pulled into common define

**Comment by HelenCampbell** [comment]
@DavidS Updated :)

File: quarterly_report.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+created = 0
+merged = 0
+closed = 0
+
+all_pulls.each do |pull|
+  if (!puppet_members.include?(pull.user[:login]))
+    if pull[:created_at] > quarter_begin and pull[:created_at] < quarter_end
+      created = created + 1
+    end
+    #require 'pry'; binding.pry
```
Comment (DavidS) [inline quarterly_report.rb:78]: debug leftovers

File: quarterly_report.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  pr_cache.each do |pr|
+    all_pulls.push (pr[:pull])
+  end
+end
+
+puppet_members = ["bmjen", "HelenCampbell", "hunner", "DavidS", "tphoney", "jonnytpuppet", "garethr", "gregohardy", "mentat"]
+quarter_begin = Date.new(2015,11,01)
+quarter_end = Date.new(2016,01,31)
+quarter_begin = quarter_begin.to_time
+quarter_end = quarter_end.to_time
```
Comment (DavidS) [inline quarterly_report.rb:68]:
```
irb(main):004:0> Date.new(2016,01,31).to_time
=> 2016-01-31 00:00:00 +0000
irb(main):005:0> 
```

that is 24 hours too early.

File: quarterly_report.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  pr_cache = (util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'open'}, filter=[]))
+  pr_cache.concat(util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'closed'}, filter=[]))
+
+  pr_cache.each do |pr|
+    all_pulls.push (pr[:pull])
+  end
+end
+
+puppet_members = ["bmjen", "HelenCampbell", "hunner", "DavidS", "tphoney", "jonnytpuppet", "garethr", "gregohardy", "mentat"]
+quarter_begin = Date.new(2015,11,01)
```
Comment (DavidS) [inline quarterly_report.rb:65]: Might be good to add a note that these are business' quarters, and therefore crooked. Just as a reminder for folks outside PL

**Comment by tphoney** [comment]
@HelenCampbell what's the status of this ? do you need a hand ?

**Comment by HelenCampbell** [comment]
@tphoney I completely forgot about this. You can play about with it if you want. If you aren't bothered I'll try and fit in sorting it out with Davids comments for merge. Isn't important, but it is interesting to see numbers per quarter.

**Pull request closed without merging**